gdal/port/cpl_vsil.cpp

```diff
@@ -437,7 +437,7 @@ void VSICloseDir(VSIDIR* dir)
  * \brief Create a directory.
  *
  * Create a new directory with the indicated mode.  The mode is ignored
- * on some platforms.  A reasonable default mode value would be 0666.
+ * on some platforms.  A reasonable default mode value would be 0777.
```

Comment:
What do you think of
```suggestion
 * on some platforms.  A reasonable default mode value would be 0777 to grant
 read, write, execute permissions to user, group and other. Or 0755 for read,
 write, execute permissions to user, and only read and execute permissions to group
 and other. On some filesystems, the umask mechanism may further restrict those permissions.
```

Reply:
I would like to keep it simple, and I don't think having two recommended value serves this purpose. (Why not 0775?) If in doubt, stick with 0755. Everything is better than 0666 (which gave me GDAL crashes in subsequent file access).

But looking at the various `mkdir` man pages, I now think the first sentense needs to be rectified, too: As explained, the function will not create the dir "with the indicated mode". So:

~~~
Create a new directory. The indicated mode is modified by
the file creation mask (umask) on POSIX-style systems,
so a reasonable default mode value is 0777. The mode is
ignored on some platforms.
~~~


We could also wait for others to comment.